junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertIterableEquals.java

```diff
@@ -71,45 +71,45 @@ private static void assertIterableEquals(Iterable<?> expected, Iterable<?> actua
 			}
 
 			indexes.addLast(processed - 1);
-			assertIterableElementsEqual(expectedElement, actualElement, indexes, messageSupplier);
+			assertIterableElementsEqual(expectedElement, actualElement, indexes, messageContainer);
 			indexes.removeLast();
 		}
 
-		assertIteratorsAreEmpty(expectedIterator, actualIterator, processed, indexes, messageSupplier);
+		assertIteratorsAreEmpty(expectedIterator, actualIterator, processed, indexes, messageContainer);
 	}
 
 	private static void assertIterableElementsEqual(Object expected, Object actual, Deque<Integer> indexes,
-			Supplier<String> messageSupplier) {
+			Object messageContainer) {
 		if (expected instanceof Iterable && actual instanceof Iterable) {
-			assertIterableEquals((Iterable<?>) expected, (Iterable<?>) actual, indexes, messageSupplier);
+			assertIterableEquals((Iterable<?>) expected, (Iterable<?>) actual, indexes, messageContainer);
 		}
 		else if (!Objects.equals(expected, actual)) {
-			assertIterablesNotNull(expected, actual, indexes, messageSupplier);
-			failIterablesNotEqual(expected, actual, indexes, messageSupplier);
+			assertIterablesNotNull(expected, actual, indexes, messageContainer);
+			failIterablesNotEqual(expected, actual, indexes, messageContainer);
 		}
 	}
 
 	private static void assertIterablesNotNull(Object expected, Object actual, Deque<Integer> indexes,
-			Supplier<String> messageSupplier) {
+			Object messageContainer) {
 
 		if (expected == null) {
-			failExpectedIterableIsNull(indexes, messageSupplier);
+			failExpectedIterableIsNull(indexes, messageContainer);
 		}
 		if (actual == null) {
-			failActualIterableIsNull(indexes, messageSupplier);
+			failActualIterableIsNull(indexes, messageContainer);
 		}
 	}
 
-	private static void failExpectedIterableIsNull(Deque<Integer> indexes, Supplier<String> messageSupplier) {
-		fail(buildPrefix(nullSafeGet(messageSupplier)) + "expected iterable was <null>" + formatIndexes(indexes));
+	private static void failExpectedIterableIsNull(Deque<Integer> indexes, Object messageContainer) {
+		fail(buildPrefix(nullSafeGet(messageContainer)) + "expected iterable was <null>" + formatIndexes(indexes));
 	}
 
-	private static void failActualIterableIsNull(Deque<Integer> indexes, Supplier<String> messageSupplier) {
-		fail(buildPrefix(nullSafeGet(messageSupplier)) + "actual iterable was <null>" + formatIndexes(indexes));
+	private static void failActualIterableIsNull(Deque<Integer> indexes, Object messageContainer) {
+		fail(buildPrefix(nullSafeGet(messageContainer)) + "actual iterable was <null>" + formatIndexes(indexes));
 	}
 
 	private static void assertIteratorsAreEmpty(Iterator<?> expected, Iterator<?> actual, int processed,
```

Comment:
@sbrannen this method and some others require to be copy/pasted if there is no `nullSafeGet(Object)` utility method. What do you think?

Reply:
Same as above... I'll have to review it all together on my machine to come to a conclusion.

So just leave it "as is" for now.